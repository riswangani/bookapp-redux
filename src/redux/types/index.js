export const LOGIN_USER = '@LOGIN_USER';
export const REGIST_USERS = '@REGIST_USERS';
export const GET_BOOKS = '@GET_BOOKS';
export const GET_DETAIL_BOOK = '@GET_DETAIL_BOOK';
