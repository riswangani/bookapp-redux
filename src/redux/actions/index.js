import axios from 'axios';

import {LOGIN_USER, REGIST_USERS} from '../types';
const registerUrl = 'http://code.aldipee.com/api/v1';

export const userRegister = (name, email, password, {navigation}) => {
  return dispatch => {
    axios
      .post('http://code.aldipee.com/api/v1/auth/register', {
        name,
        email,
        password,
      })
      .then(res => {
        const resRegist = res.data;
        dispatch({type: LOGIN_USER, payload: resRegist.data});
        console.log(resRegist);
        if (resRegist.success) {
          navigation.replace('SuccessRegister');
        }
      });
  };
};

export const userLogin = (email, password, {navigation}) => {
  return dispatch => {
    axios.post(registerUrl + '/auth/login', {email, password}).then(res => {
      const resLogin = res.data;
      dispatch({type: LOGIN_USER, payload: resLogin.data});
      if (resLogin.user.isEmailVerified) {
        const namauser = resLogin.user.name;
        const barier = resLogin.tokens.access.token;
        axios
          .get(`${registerUrl}/books`, {
            headers: {
              Authorization: `Bearer ` + barier,
            },
          })
          .then(resBook => {
            const dataBook = resBook.data;
            navigation.navigate('Home', {
              data: dataBook,
              namaUser: namauser,
              barier: barier,
            });
          });
      }
    });
  };
};
