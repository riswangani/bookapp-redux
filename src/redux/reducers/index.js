import {combineReducers} from 'redux';
import {REGIST_USERS, GET_BOOKS, LOGIN_USER} from '../types';

// register
const initialRegisterState = {
  form: {},
};

const registerReducer = (state = initialRegisterState, action) => {
  switch (action.type) {
    case REGIST_USERS:
      return {
        ...state,
        form: action.payload,
      };
    default:
      return state;
  }
};

// login
const initialLoginState = {
  form: {},
};

const loginReducer = (state = initialLoginState, action) => {
  switch (action.type) {
    case LOGIN_USER:
      return {
        ...state,
        form: action.payload,
      };
    default:
      return state;
  }
};

const reducer = combineReducers({
  registerReducer,
  loginReducer,
});

export default reducer;
