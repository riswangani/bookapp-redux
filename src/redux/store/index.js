import thunk from 'redux-thunk';
import {createStore, combineReducers, applyMiddleware} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';
import reducer from '../reducers';

// const rootReducer = combineReducers({
//   reducer,
// });

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const configPersist = persistReducer(persistConfig, reducer);

export const Store = createStore(configPersist, applyMiddleware(thunk));
export const PersistStore = persistStore(Store);
