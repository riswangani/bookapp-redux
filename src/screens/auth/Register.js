import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  Image,
  ScrollView,
} from 'react-native';
import React, {useState} from 'react';

import {useDispatch, useSelector} from 'react-redux';

import {notatLogo} from '../../assets/images';
import CustomInput from '../../components/CustomInput/CustomInput';
import CustomButton from '../../components/CustomButton/CustomButton';
import CustomButtonBack from '../../components/CustomButton/CustomButtonBack';
import {userRegister} from '../../redux';
import axios from 'axios';

const Register = ({navigation}) => {
  // const {form} = useSelector(state => state.registerReducer);

  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const dispatch = useDispatch();

  // const onInputChange = (value, inputType) => {
  //   dispatch(userRegister(inputType, value));
  // };

  const onSignUpPressed = async () => {
    dispatch(userRegister(name, email, password, {navigation}));
  };

  return (
    <View style={styles.root}>
      <ScrollView showsVerticalScrollIndicator={false}>
        <CustomButtonBack type="regis" navigation={navigation} />
        <View style={{alignItems: 'center'}}>
          <Image source={notatLogo} style={styles.logo} resizeMode="contain" />

          <CustomInput
            testID="InputName"
            placeholder="Full Name"
            onChangeText={setName}
          />
          <CustomInput
            testID="InputEmail"
            placeholder="Email"
            onChangeText={setEmail}
          />
          <CustomInput
            testID="InputPassword"
            placeholder="Password"
            onChangeText={setPassword}
            secureTextEntry={true}
          />
          <CustomButton
            testID="myButtonRegist"
            text="Register"
            onPress={onSignUpPressed}
          />

          <Text>Already Have an Account?</Text>

          <CustomButton
            text="Login"
            onPress={() => navigation.navigate('Login')}
            type="TERTIARY"
          />
        </View>
      </ScrollView>
    </View>
  );
};

export default Register;

const windowHeight = Dimensions.get('window').height;

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: '#ffffff',
    padding: 20,
  },
  logo: {
    width: '70%',
    maxWidth: 300,
    maxHeight: 200,
    height: windowHeight * 0.2389162562,
  },
});
