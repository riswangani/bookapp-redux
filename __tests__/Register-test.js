import 'react-native';
import React from 'react';
import {create, act, renderer} from 'react-test-renderer';
import {createStore, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../src/redux/reducers';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';

import Register from '../src/screens/auth/Register';

const Store = createStore(reducer, applyMiddleware(thunk));

configure({adapter: new Adapter()});

const tree = create(
  <Provider store={Store}>
    <Register />
  </Provider>,
);

describe('RegisterScreen', () => {
  it('should render Register Screen, module correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('renders button registrasi', () => {
    const inputName = tree.root.findByProps({testID: 'InputName'}).props;
    const inputEmail = tree.root.findByProps({testID: 'InputEmail'}).props;
    const inputPassword = tree.root.findByProps({
      testID: 'InputPassword',
    }).props;
    const button = tree.root.findByProps({testID: 'myButtonRegist'}).props;
    act(() => inputName.onChangeText('Riswan Gani Padilah'));
    act(() => inputEmail.onChangeText('riswan.qwe123@gmail.com'));
    act(() => inputPassword.onChangeText('Riswan12344'));
    act(() => button.onPress());
  });
});
