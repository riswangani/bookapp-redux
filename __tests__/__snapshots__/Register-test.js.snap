// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen should render Register Screen, module correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "marginVertical": 5,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï„©
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 318.7142857708,
              "maxHeight": 200,
              "maxWidth": 300,
              "width": "70%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginVertical": 5,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Full Name"
            style={Object {}}
            testID="InputName"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginVertical": 5,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Email"
            style={Object {}}
            testID="InputEmail"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginVertical": 5,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={Object {}}
            testID="InputPassword"
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B7343D",
              "borderRadius": 5,
              "marginVertical": 15,
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                },
                undefined,
              ]
            }
          >
            Register
          </Text>
        </View>
        <Text>
          Already Have an Account?
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 15,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                },
                Object {
                  "color": "gray",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
